File: algorithms/Searching/Easy/LinearSearch/VSDLinearSearch.java


public class Main{
    
    static int linearSearch(int arr[],int n,int target){
        for(int i=0;i<n;i++){
            if(arr[i]==target){
                System.out.println("Found at index "+i);
                return i;
            }

        }
        System.out.println("Not found");
        return -1;
    }
    
    public static void main(String[] args) {
        int[] arr={87,48,464,16,54,18,87,87,684};
        int n=arr.length;
        linearSearch(arr,n , 54);
        linearSearch(arr, n, 0);
    }
}

File: algorithms/Searching/Easy/LinearSearch/VSDLinearSearch.cpp

#include<bits/stdc++.h>

using namespace std;

int linearSearch(int arr[],int n,int target){
    for(int i=0;i<n;i++){
        if(arr[i]==target){
            cout<<"Found at index "<<i<<endl;
            return;
        }
    }
    cout<<"Not found"<<endl;
    return -1;
}

int main(){
    int arr[]{25,49,879,1984,894,6879,3419,354,165};
    int n=sizeof(arr)/sizeof(arr[0]);
    linearSearch(arr,n,894);
    linearSearch(arr,n,0);
    return 0;
}